### ISSUE TITLE
SHACL SPARQL Mapping of Variables to sh:message

### ISSUE THREAD

**Issue by florianschmi** [RDFLib/pySHACL#120]
Hi,
I would like to display variables of the SPARQL select query in the sh:message like specified in the [SHACL documentation](https://www.w3.org/TR/shacl/#sparql-constraints-variables). Unfortunately, it does not work like expected.

How can I display variables of the SPARQL select query in the sh:message? Is there any workaround?
Thanks for your support.

Example SHACL:

```
maic:RuleShapeI rdf:type owl:NamedIndividual , sh:NodeShape ;
 sh:sparql [
  rdf:type sh:SPARQLConstraint ;
  sh:message "{$this} is a teacher of {?student}." ;
  sh:prefixes [
   sh:declare [
    sh:namespace "http://www.semanticweb.org/anyNamespace"^^xsd:anyURI ;
    sh:prefix "maic"
   ]
  ] ;
 sh:select """
  SELECT $this ?student
  WHERE {
  $this maic:teaches ?student
  }"""
 ] ;
 sh:targetClass maic:Teacher ;
 sh:severity sh:Info ;
.
```

Example validation report:

`sh:Message "{$this} is a teacher of {?student}." .`

Expected validation report:

`sh:Message "Anna is a teacher of Boris." .`

**Comment by ashleysommer** [comment]
Hi @florianschmi 
I thought I had already implemented this feature, but looks like it might be in a branch that is not merged. I will check to see if I still have the code, and try to get the feature working for you.

**Comment by florianschmi** [comment]
Hi @ashleysommer,
Thank you for the answer. Could you already find and merge the branch mentioned?
In the meanwhile, I managed to display the $this variable in the SHACL Message properly. However, other variables than $this still don't seem to work.

Example validation report:
`sh:Message "Anna is a teacher of {?student}." .`

**Comment by ashleysommer** [comment]
Hi @florianschmi 
Sorry, I've been very busy and haven't had time to look into this yet.

>  I managed to display the $this variable in the SHACL Message properly.

That's right. All "pre-bound" variables are be available in the result message already, that is `?this` `?shapesGraph` `?currentShape`, plus I think `?value` and `?path`.

**Comment by florianschmi** [comment]
Hi @ashleysommer,

I understand. Still, I would be very grateful if you could find the time.
I appreciate any help you can provide.

**Comment by ashleysommer** [comment]
@florianschmi 
This feature is finally re-implemented, and has been released in the latest PySHACL v0.19.0
Can you please check it out and let me know if it works for you.

**Comment by florianschmi** [comment]
Hi @ashleysommer,
It works perfectly. Thank you so much for the support and development.
